### PR TITLE
bulk deletion, prompt image etc.

### PR DESCRIPTION
**Pull request by 12343954**
## some features 

- 🆒 Bulk deletion
- 🔥 Displays the image of every prompt, easy to preview, apply or delete
- 💖 Fix size of the preview image
- 🛶 Adapt to page layout, below 2k resolution (perfect in 4k)

## code side
- 📐 Optimize web-side code performance
- 🧹 Format the code for easy maintenance

### CODE REVIEW

**Comment by 12343954** [comment]
sorry，please ignore the `Readme.md`, not very good at `PR`.

**Comment by namkazt** [comment]
> sorry，please ignore the `Readme.md`, not very good at `PR`.

can you remove the readme file from PR or update the features part only?

File: javascript/prompt_history.js [hunk 1]
```diff
@@ -41,69 +41,104 @@ class ClassWatcher {
...
+
+    let textarea = document.querySelector('#prompt_history_item_id_text textarea');
+    textarea.value = Array.from(checkboxes).map(p => p.value).join(',');
+    updateInput(textarea);
+    document.querySelector('#prompt_history_delete_item_btn').click();
+}
+
 function promptHistoryItemClick(id) {
-    var textarea = gradioApp().querySelector('#prompt_history_item_id_text textarea');
+    let textarea = document.querySelector('#prompt_history_item_id_text textarea');
```
Comment (namkazt) [inline javascript/prompt_history.js:69]: i still prefer `gradioApp()` in here

File: scripts/prompt_history_script.py [hunk 2]
```diff
@@ -67,89 +69,109 @@ def add_config(id: str, name: str, model: str, info_text: str, img) -> history.H
...
-    
+
     # save image
     if global_state.save_thumbnail:
-        new_width  = 300
+        new_width = 300
         new_height = int(new_width * img.height / img.width)
         img = img.resize((new_width, new_height), Image.LANCZOS)
-        images.save_image_with_geninfo(img, None, os.path.join(global_state.history_path, f"{id}.jpg"))
+        images.save_image_with_geninfo(img, None, os.path.join(
```
Comment (namkazt) [inline scripts/prompt_history_script.py:78]: Not sure what everyone thinking but with me I think this is harder to read. xD

**Comment by 12343954** [comment]
> > sorry，please ignore the `Readme.md`, not very good at `PR`.
> 
> can you remove the readme file from PR or update the features part only?

how to remove it? I just know some simple options in github.

File: scripts/prompt_history_script.py (hunk 2 above)
Comment (12343954) [reply scripts/prompt_history_script.py:78]: vs code `auto-format` here, you can ignore it, and use your prefer format.

File: javascript/prompt_history.js (hunk 1 above)
Comment (12343954) [reply javascript/prompt_history.js:69]: use what you like.

**Comment by namkazt** [comment]
> > > sorry，please ignore the `Readme.md`, not very good at `PR`.
> > 
> > 
> > can you remove the readme file from PR or update the features part only?
> 
> how to remove it? I just know some simple options in github.

maybe I will patch the current source from your repo and create a reference to your MR. it will be easier to do.

**Comment by 12343954** [comment]
> > > > sorry，please ignore the `Readme.md`, not very good at `PR`.
> > > 
> > > 
> > > can you remove the readme file from PR or update the features part only?
> > 
> > 
> > how to remove it? I just know some simple options in github.
> 
> maybe I will patch the current source from your repo and create a reference to your MR. it will be easier to do.

That's okay, just do it the easiest way. github is too hard to use, I don't like it. 

Maybe you merge the code, restore the old code, and commit it again. 
I often do this on SVN, but don't know if it's okay to do this on github.

**Comment by namkazt** [comment]
> > > > > sorry，please ignore the `Readme.md`, not very good at `PR`.
> > > > 
> > > > 
> > > > can you remove the readme file from PR or update the features part only?
> > > 
> > > 
> > > how to remove it? I just know some simple options in github.
> > 
> > 
> > maybe I will patch the current source from your repo and create a reference to your MR. it will be easier to do.
> 
> That's okay, just do it the easiest way. github is too hard to use, I don't like it.
> 
> Maybe you merge the code, restore the old code, and commit it again. I often do this on SVN, but don't know if it's okay to do this on github.

i was used svn for a very long time ago :D at least 12 years for sure. 

anyway i will handle this case. thanks for the PR

**Comment by namkazt** [comment]
thanks for the PR, all ideas are implemented.

**Pull request closed without merging**